src/server/services/user.service.ts

```diff
@@ -534,32 +534,31 @@ export const removeAllContent = async ({ id }: { id: number }) => {
     select: { id: true },
   });
 
-  const res = await Promise.all([
-    dbWrite.model.deleteMany({ where: { userId: id } }),
-    dbWrite.comment.deleteMany({ where: { userId: id } }),
-    dbWrite.commentV2.deleteMany({ where: { userId: id } }),
-    dbWrite.resourceReview.deleteMany({ where: { userId: id } }),
-    dbWrite.post.deleteMany({ where: { userId: id } }),
-
-    dbWrite.article.deleteMany({ where: { userId: id } }),
-    dbWrite.userProfile.deleteMany({ where: { userId: id } }),
-    dbWrite.userLink.deleteMany({ where: { userId: id } }),
-    dbWrite.question.deleteMany({ where: { userId: id } }),
-    dbWrite.answer.deleteMany({ where: { userId: id } }),
-    dbWrite.collection.deleteMany({ where: { userId: id } }),
-    dbWrite.bounty.deleteMany({ where: { userId: id } }),
-    dbWrite.bountyEntry.deleteMany({
-      where: { userId: id, benefactors: { none: {} } },
-    }),
-    dbWrite.imageReaction.deleteMany({ where: { userId: id } }),
-    dbWrite.articleReaction.deleteMany({ where: { userId: id } }),
-    dbWrite.commentReaction.deleteMany({ where: { userId: id } }),
-    dbWrite.commentV2Reaction.deleteMany({ where: { userId: id } }),
-    // TODO - an implementation similar to the following
-    // loop through image
-    // imageService.deleteImageById
-    dbWrite.image.deleteMany({ where: { userId: id } }),
-  ]);
+  // sort deletes by least impactful to most impactful
+  await dbWrite.imageReaction.deleteMany({ where: { userId: id } });
+  await dbWrite.articleReaction.deleteMany({ where: { userId: id } });
+  await dbWrite.commentReaction.deleteMany({ where: { userId: id } });
+  await dbWrite.commentV2Reaction.deleteMany({ where: { userId: id } });
+  await dbWrite.bountyEntry.deleteMany({
+    where: { userId: id, benefactors: { none: {} } },
+  });
+  await dbWrite.bounty.deleteMany({ where: { userId: id } });
+  await dbWrite.answer.deleteMany({ where: { userId: id } });
+  await dbWrite.question.deleteMany({ where: { userId: id } });
+  await dbWrite.userLink.deleteMany({ where: { userId: id } });
+  await dbWrite.userProfile.deleteMany({ where: { userId: id } });
+  await dbWrite.resourceReview.deleteMany({ where: { userId: id } });
+  await dbWrite.commentV2.deleteMany({ where: { userId: id } });
+  await dbWrite.comment.deleteMany({ where: { userId: id } });
+  await dbWrite.collection.deleteMany({ where: { userId: id } });
+  await dbWrite.article.deleteMany({ where: { userId: id } });
+  await dbWrite.post.deleteMany({ where: { userId: id } });
+  await dbWrite.model.deleteMany({ where: { userId: id } });
+
+  // TODO - an implementation similar to the following
+  // loop through image
+  // imageService.deleteImageById
+  await dbWrite.image.deleteMany({ where: { userId: id } });
```

Comment:
Might be a good idea to wrap all of this in a try/catch block since we call it directly in the `user.router` file. Or we can have a controller for it and wrap it all there to keep consistency.

Reply:
I'm pretty sure we don't need to wrap anything in a try catch block anywhere (even a controller) unless we want additional error handling. We call `handleTRPCError` in `[trpc].ts`. Any error handling being done in a controller should automatically be handled by this handler.